test/spec/unit/remote.js

```diff
@@ -157,5 +158,104 @@ describe('remote method', () => {
 
             return assert.isRejected(client.getUrl(), /some-error/)
         })
+
+        it('should not change connection retry options if "screenshotOnReject" is boolean', () => {
+            var client = remote({screenshotOnReject: true})
+            var {connectionRetryTimeout, connectionRetryCount} = client.options
+
+            return client.getUrl()
+                .catch(() => {
+                    assert.strictEqual(client.options.connectionRetryTimeout, connectionRetryTimeout)
+                    assert.strictEqual(client.options.connectionRetryCount, connectionRetryCount)
+                })
+        })
+
+        it('should set default connection retry options if "screenshotOnReject" is empty object', () => {
+            var client = remote({screenshotOnReject: {}})
+            var defaultConnectionOpts = {
+                connectionRetryTimeout: client.options.connectionRetryTimeout,
+                connectionRetryCount: client.options.connectionRetryCount
+            }
+
+            return client.getUrl()
+                .catch(() => assert.calledWith(RequestHandler.prototype.updateConnectionOpts, defaultConnectionOpts))
+        })
+
+        it('should set specified connection retry options', () => {
+            var connectionOpts = {
+                connectionRetryTimeout: 30000,
+                connectionRetryCount: 0
+            }
+            var client = remote({screenshotOnReject: connectionOpts})
+
+            return client.getUrl()
+                .catch(() => assert.calledWith(RequestHandler.prototype.updateConnectionOpts, connectionOpts))
+        });
+
+        ['completed successfully', 'failed'].forEach((message) => {
+            it(`should reset connection retry options to default if takeScreenshot is ${message}`, () => {
+                var client = remote({
+                    screenshotOnReject: {
+                        connectionRetryTimeout: 30000,
+                        connectionRetryCount: 0
+                    }
+                })
+
+                var defaultConnectionRetryTimeout = client.options.connectionRetryTimeout
+                var defaultConnectionRetryCount = client.options.connectionRetryCount
```

Comment:
In fact we can't be sure that default values are not the 3000 and 0. Lets set those options to some values and then check that values